### PR TITLE
DAOS-10739 dfuse: Increase timeouts for daos_build test.

### PR DESCRIPTION
**Pull request by ashleypittman**
Change timeouts and settings for daos_build test so that it works
again.  A recent change in VM configuration increased the runtime
so this test was failing and as a result the test got disabled.
Increase the timeouts so the test passes again, and move the test
back to daily running.

Signed-off-by: Ashley Pittman <ashley.m.pittman@intel.com>

### CODE REVIEW

**Comment by ashleypittman** [comment]
@kccain any thoughts on this and scm memory sizes?

**Comment by kccain** [comment]
> @kccain any thoughts on this and scm memory sizes?
I don't imagine there is a problem with oversubscription of memory in the VMs based on this change. Based on below analysis.

With this patch the pool size 12GiB is being divided over the 6 server VMs so each pool would be taking only 2GB of "scm" (tmpfs emulation in the case of a VM). This is due to the pool create command using the "--size=" argument (as opposed to --scm-size= that would allocate the specified amount on a per-engine basis). The /mnt/daos partition tmpfs capacity seems to be 16GB with 2.2GB in use (I peeked at the test run in progress on wolf-106vm* to confirm pool sizes and partition sizes)

Before this change it looks like 5GiB of scm was allocated for the pool on a single server VM (via --size=5GiB).

I've seen some other VM-based testing have issues with a different ("scm_size") parameter (defined either underneath server_config/servers or "pool" in the test .yaml). Cases such as scm_size: 8 compared to VM instances in CI being given 12GB of System RAM, not leaving enough room for a daos_engine instance to run effectively without (presumably) encountering paging/swapping, or (observed) getting killed by the Linux oom-killer in some situations.

**Comment by ashleypittman** [comment]
> I've seen some other VM-based testing have issues with a different ("scm_size") parameter (defined either underneath server_config/servers or "pool" in the test .yaml). Cases such as scm_size: 8 compared to VM instances in CI being given 12GB of System RAM, not leaving enough room for a daos_engine instance to run effectively without (presumably) encountering paging/swapping, or (observed) getting killed by the Linux oom-killer in some situations.

I'd heard you had seen issues like that and I'm glad it didn't apply here.

we've seen a recent 6x performance regression in this test, the runtime has gone from 13 minutes to one hour fourty five minutes and we're not sure why.  This PR is an attempt to resolve it, moving to more servers with more overall space but less per server was an attempt to improve things but it doesn't seem to have helped.

**Comment by kccain** [comment]
> we've seen a recent 6x performance regression in this test, the runtime has gone from 13 minutes to one hour fourty five minutes and we're not sure why. This PR is an attempt to resolve it, moving to more servers with more overall space but less per server was an attempt to improve things but it doesn't seem to have helped.

The other thing I've tried experimentally with some success is (in the test .yaml config file) set the engine configuration crt_timeout: higher than the functional testing default 10 seconds that was put in place with commit[4c4cd51]( https://github.com/daos-stack/daos/commit/4c4cd51a405d2849521380075fdd9fa3e1920627) . For example setting it to 20 for one test that (pool/create_all_vm) was having some RPC timeout and retry symptoms. I didn't commit any crt_timeout changes to tests, instead focused on adjusting the scm_size parameter for a few tests running in the VM environment.

File: src/tests/ftest/dfuse/daos_build.py [hunk 1]
```diff
@@ -111,22 +113,26 @@ def test_daos_build(self):
...
 
         preload_cmd = ';'.join(envs)
 
+        build_jobs = 6 * 5
+        if intercept:
+            build_jobs = 1
+
         # Run the deps build in parallel for speed/coverage however the daos build itself does
-        # not yet work, so run this part in serial.
+        # not yet work, so run this part in serial.  The VMs have 6 cores each.
```
Comment (mchaarawi) [inline src/tests/ftest/dfuse/daos_build.py:127]: can we do some auto-detection of core count? this sounds site specific and would have to change if one reconfigures the VM? or runs on HW?

**Review by mchaarawi — APPROVED** [review]

File: src/tests/ftest/dfuse/daos_build.yaml [hunk 2]
```diff
@@ -3,10 +3,14 @@ hosts:
     - server-A
   test_clients:
     - client-C
-# Hard limit the test at three hours, however individual tests have lower timeouts.
-timeout: 10800
+# Hard limit the test at six hours, however individual tests have lower timeouts.
+timeout: 21600
```
[the remark this answers is not in the document]
Comment (ashleypittman) [reply src/tests/ftest/dfuse/daos_build.yaml:7]: Unless we actually run this test then I don't know.  Down from 17 hours which but seems to be about six in total, although I believe this setting applies to the elements which the longest is just under three.

https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-9389/20/testReport/FTEST_dfuse/DaosBuild/

The intent is to keep working on this once re-enabled for dailies in the hope we can get this down and resolve the remaining intermittent issues with it, plus there's the fact that we keep landing regressions to dfuse which makes re-enabling this test harder every time.

[the remark this answers is not in the document]
Comment (ashleypittman) [reply src/tests/ftest/dfuse/daos_build.yaml:7]: Alternatively we could land the rest of this PR but keep it at weekly, that would at least get Test-tag: dfuse working again which would solve a lot of the pain I'm experiencing currently.

**Pull request merged**